src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -116,23 +117,70 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 		}
 	}
 
-	public static transformData(rows: string[][], columns: any[]): { [key: string]: string }[] {
+	private static createIconColumn<T extends Slick.SlickData>(col: azdata.TableColumn): Slick.Column<T> {
+		return <Slick.Column<T>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: iconCssFormatter,
+			filterable: false
+		};
+	}
+
+	private static createTextColumn(col: azdata.TableColumn): Slick.Column<any> {
+		return <Slick.Column<any>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: textFormatter,
+		};
+	}
+
+
+
+	public static transformData(rows: (string | azdata.IconColumnCellValue)[][], columns: any[], iconCssMap?: { [iconKey: string]: string }): { [key: string]: string | CssIconCellValue }[] {
```

Comment:
Instead of passing in the map why not just make this a member of the class - that'd simplify it

Reply:
There are some tests for this method. Don't want to change them a lot.

Reply:
Although, this is not a big change. Implemented.